### PR TITLE
[IN-113] Feat/in 113 view update

### PR DESCRIPTION
**Pull request by ghost**
#### User Story

- https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-113

#### Changes made

(Summary of messenger text):

- There was a race condition proving problematic when it came to rendering the feedback information our View page. 
- I realized that the matches back-end isn't really being used for any functional purpose beyond the get request in View. 
- I chose to make a design decision and expand the JobPostingSchema to include a new "notes" field. 
- I created a new students field to contain an array of studentIDs, rather than simply the number.
- Our functionality is preserved. Deleting a jobPosting will now, for all intents and purposes, delete all the relevant match information with it (no need for two delete calls). 
- In addition, for now, I will keep the matches route/model and so on. Perhaps it will be useful in the future. 

Also, in future, for this merge to work with edit, ideally edit is merged first to develop. I'll then merge edit into this, and make the necessary endpoint changes for editing to work with the new schema. (PUT endpoint to accommodate two new fields, and an additional function before sending, that grabs these fields from reducer, and places them into the jobObject being sent)

#### Does your new code introduce new warnings on dev console?
No.

#### Test Steps
**Create a new job! Then click on it to view!** Do not try to edit, or view old jobs (there will be an issue if you click the eye on old jobs)

### CODE REVIEW

**Review by tommytwm — APPROVED** [review]
There is a slight bug where the delete button shifts to the same row as the other buttons after toggling the Feedback column, but that's fine for now.

File: backend/models/JobPostingData.js [hunk 1]
```diff
@@ -18,14 +18,17 @@ const JobPostingSchema = new Schema({
...
+  students: {
+    type: [String],
+  },
   dateCreated: {
     type: Date,
   },
   dateUpdated: {
     type: Date,
   },
-  matches: {
```
Comment (JudeShamsi) [inline backend/models/JobPostingData.js:27]: why are we removing matches? we need this to display on the profile page I thought

File: backend/routes/api/jobs.js [hunk 2]
```diff
@@ -67,7 +69,8 @@ router.put("/:id", function (req, res, next) {
     requirements: req.body.requirements,
     details: req.body.details,
     contact: req.body.contact,
-    matches: req.body.matches,
```
Comment (JudeShamsi) [inline backend/routes/api/jobs.js:70]: why is this deleted

File: backend/models/JobPostingData.js (hunk 1 above)
Comment (tommytwm) [reply backend/models/JobPostingData.js:27]: This is a duplicate field that he's removing

File: backend/routes/api/jobs.js (hunk 2 above)
Comment (tommytwm) [reply backend/routes/api/jobs.js:70]: Ya why is this deleted

File: client/src/components/pages/Edit.js [hunk 3]
```diff
@@ -27,13 +31,45 @@ import "firebase/auth";
...
+    "Node",
+    "React",
+    "Ruby on Rails",
+    "Unix",
+  ],
+  workTools: ["Azure", "GitHub", "Jira", "Jupyter"],
   csConcepts: [
-    "Object Oriented Programming",
+    "Agile Development",
+    "Algorithms",
```
Comment (tommytwm) [inline client/src/components/pages/Edit.js:64]: What Algorithms? Search Algorithms? Sorting Algorithms?

File: client/src/components/pages/Edit.js [hunk 4]
```diff
@@ -27,13 +31,45 @@ import "firebase/auth";
...
+    "Asynchronous Programming",
+    "Data Structures",
+    "Design Principles & Patterns",
     "Functional Programming",
+    "Object Oriented Programming",
+    "RESTify Services",
```
Comment (tommytwm) [inline client/src/components/pages/Edit.js:70]: `REST` will do

File: client/src/components/pages/Edit.js [hunk 5]
```diff
@@ -27,13 +31,45 @@ import "firebase/auth";
...
     "Recursion",
+    "Web APIs",
```
Comment (tommytwm) [inline client/src/components/pages/Edit.js:72]: `REST` would fall under `Web APIs` you can remove this unless you know which particular API's you're looking for — which would then typically fall under frameworks

File: client/src/components/pages/Edit.js [hunk 6]
```diff
@@ -27,13 +31,45 @@ import "firebase/auth";
...
+    "Unix",
+  ],
+  workTools: ["Azure", "GitHub", "Jira", "Jupyter"],
   csConcepts: [
-    "Object Oriented Programming",
+    "Agile Development",
+    "Algorithms",
+    "Asynchronous Programming",
+    "Data Structures",
+    "Design Principles & Patterns",
```
Comment (tommytwm) [inline client/src/components/pages/Edit.js:67]: What kind of Design Principles? Graphic designers have their own design principles. Might be ideal to add in `Software` in front of this

**Review by tommytwm — CHANGES_REQUESTED** [review]

File: backend/routes/api/jobs.js (hunk 2 above)
Comment (ghost) [reply backend/routes/api/jobs.js:70]: 👀I just changed the order, trust me its there

File: client/src/components/pages/Edit.js (hunk 3 above)
Comment (ReneReid) [reply client/src/components/pages/Edit.js:64]: per conversation with @tommytwm  - this has been resolved

File: client/src/components/pages/Edit.js (hunk 6 above)
Comment (ReneReid) [reply client/src/components/pages/Edit.js:67]: per conversation with @tommytwm  this has been resolved

File: client/src/components/pages/Edit.js (hunk 4 above)
Comment (ReneReid) [reply client/src/components/pages/Edit.js:70]: per conversation with @tommytwm this has been resolved

File: client/src/components/pages/Edit.js (hunk 5 above)
Comment (ReneReid) [reply client/src/components/pages/Edit.js:72]: per conversation with @tommytwm this has been resolved

**Review by tommytwm — APPROVED** [review]

File: client/src/effects/filter.effects.js [hunk 7]
```diff
@@ -1,102 +1,108 @@
...
+    gpaValue: jobPosting.requirements.gpaValue,
+    languages: languages,
+    frameworks: jobPosting.requirements.frameworks,
+    tools: jobPosting.requirements.tools,
+    concepts: concepts,
+    candidates: candidates,
+    academicReq: jobPosting.details.academicReq,
+    coOp: coOp,
+  };
+}
```
Comment (JudeShamsi) [inline client/src/effects/filter.effects.js:89]: does this object need to get added with the other attributes?

Comment (ghost) [reply client/src/effects/filter.effects.js:89]: Can you please clarify? These attributes are needed to be sent to the reducer, since they can be involved in filtering.

Comment (JudeShamsi) [reply client/src/effects/filter.effects.js:89]: do you need the notes, matches, profile...etc? Just wondering what this object is used for and if it requires all the job object information or not

Comment (JudeShamsi) [reply client/src/effects/filter.effects.js:89]: or is it ok to just maintain these attributes and not have all the attributes required for a job object?

**Review by JudeShamsi — APPROVED** [review]
